packageextractor.cake

```diff
@@ -0,0 +1,99 @@
+class PackageExtractor
+{
+    static PackageSaveMode packageSaveMode = PackageSaveMode.Defaultv3;
+
+    public static void Extract(string fileName)
+    {
+        string dir = Environment.CurrentDirectory;
+        Extract(fileName, dir);
+    }
+
+    public static void Extract(string fileName, string targetPath)
+    {
+        using (var stream = new FileStream(fileName, FileMode.Open, FileAccess.Read))
+        {
+            Extract(stream, targetPath);
+        }
+    }
+
+    public static void Extract(Stream nupkgStream, string targetPath)
+    {
+        var tempHashPath = System.IO.Path.Combine(targetPath, System.IO.Path.GetRandomFileName());
+        var targetNuspec = System.IO.Path.Combine(targetPath, System.IO.Path.GetRandomFileName());
+        var targetNupkg = System.IO.Path.Combine(targetPath, System.IO.Path.GetRandomFileName());
+        targetNupkg = System.IO.Path.ChangeExtension(targetNupkg, ".nupkg");
+        var hashPath = System.IO.Path.Combine(targetPath, System.IO.Path.GetRandomFileName());
+
+        using (var packageReader = new PackageArchiveReader(nupkgStream))
+        {
+            var nuspecFile = packageReader.GetNuspecFile();
+            if ((packageSaveMode & PackageSaveMode.Nuspec) == PackageSaveMode.Nuspec)
+            {
+                packageReader.ExtractFile(nuspecFile, targetNuspec, new NullLogger());
```

Comment:
Why NullLogger? there was some global context or so

Reply:
This class is from another inheritance tree 
`NuGet.Common.ILogger`.